compile error about gnutls

Cockpit version: 202
OS: CentOS 7
Page: <!--- If applicable, for instance "Network", "Machines", etc... -->

<!--- Issue description -->

<!---
Please help us out by explaining how to reproduce it.

Relevant parts of the system log are also useful:
`journalctl --since -10m` if the issue happened in the last 10 minutes
`journalctl -u cockpit -u <unit>` if you know which unit it concerns
-->

Steps to reproduce

1. make error 

make  all-am
make[1]: Entering directory `/root/src/cockpit-202'
  CC       src/tls/cockpit_tls-server.o
src/tls/server.c: In function ‘verify_peer_certificate’:
src/tls/server.c:338:18: error: ‘GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR’ undeclared (first use in this function)
           return GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR;
                  ^
src/tls/server.c:338:18: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [src/tls/cockpit_tls-server.o] error 1
make[1]: Leaving directory `/root/src/cockpit-202'
make: *** [all] error 2

2. gnutls version information
[root@centos7 tls]# rpm -qa |grep gnutls
gnutls-3.3.29-9.el7_6.x86_64
gnutls-c++-3.3.29-9.el7_6.x86_64
gnutls-dane-3.3.29-9.el7_6.x86_64
gnutls-utils-3.3.29-9.el7_6.x86_64
gnutls-devel-3.3.29-9.el7_6.x86_64

<!--- In case the issue is clearly visible, screenshots are very helpful -->

--- a/src/tls/server.c
+++ b/src/tls/server.c
@@ -335,7 +335,11 @@ verify_peer_certificate (gnutls_session_t session)
             errx (1, "Failed to print verification status: %s", gnutls_strerror (ret));
           warnx ("Invalid TLS peer certificate: %s", msg.data);
           gnutls_free (msg.data);
+#ifdef GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR
           return GNUTLS_E_CERTIFICATE_VERIFICATION_ERROR;
+#else  /* fallback for GnuTLS < 3.4.4 */
+          return GNUTLS_E_CERTIFICATE_ERROR;
+#endif
         }
     }
   else if (ret != GNUTLS_E_NO_CERTIFICATE_FOUND)

--- a/src/tls/test-server.c
+++ b/src/tls/test-server.c
@@ -471,9 +471,11 @@ test_tls_client_cert_expired (TestCase *tc, gconstpointer data)
    * handshake: that does not pick up the server's late failing handshake from the
    * verify function, only the next read attempt does */
   assert_https_outcome (tc, CLIENT_EXPIRED_CERTFILE, CLIENT_KEYFILE, 1, GNUTLS_E_SUCCESS, GNUTLS_E_PULL_ERROR);
-#else
+#elif GNUTLS_VERSION_NUMBER >= 0x030403
   /* TLS < 1.3 has a three-step handshake which picks up the invalid cert in gnutls_handshake() */
   assert_https_outcome (tc, CLIENT_EXPIRED_CERTFILE, CLIENT_KEYFILE, 1, GNUTLS_E_PULL_ERROR, 0);
+#else
+  g_test_skip ("too old GnuTLS, cannot validate certificate properties");
 #endif
 }
 

